frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,285 @@
+import { SchemaReviewId } from "./id";
+import { Principal } from "./principal";
+
+export enum RuleLevel {
+  DISABLED = "disabled",
+  ERROR = "error",
+  WARNING = "warning",
+}
+
+export const LEVEL_LIST = [
+  RuleLevel.ERROR,
+  RuleLevel.WARNING,
+  RuleLevel.DISABLED,
+];
+
+enum PayloadType {
+  STRING = "string",
+  STRING_ARRAY = "string[]",
+  TEMPLATE = "template",
+}
+
+interface StringPayload {
+  type: PayloadType.STRING;
+  default: string;
+  value?: string;
+}
+
+interface StringArrayPayload {
+  type: PayloadType.STRING_ARRAY;
+  default: string[];
+  value?: string[];
+}
+
+interface TemplatePayload {
+  type: PayloadType.TEMPLATE;
+  default: string;
+  templateList: { id: string; description?: string }[];
+  value?: string;
+}
+
+export interface RulePayload {
+  [key: string]: StringPayload | StringArrayPayload | TemplatePayload;
+}
+
+export type SchemaRuleEngineType = "MySQL" | "Common";
+
+export type CategoryType = "engine" | "naming" | "query" | "table" | "column";
+
+export interface SchemaRule {
+  id: string;
+  category: CategoryType;
+  engine: SchemaRuleEngineType;
+  description: string;
+  payload?: RulePayload;
+}
+
+export interface SelectedRule extends SchemaRule {
+  level: RuleLevel;
+}
+
+type RulePayloadType = string | string[];
+
+interface DatabaseSchemaRule {
+  id: string;
+  level: RuleLevel;
+  payload?: {
+    [key: string]: RulePayloadType;
```

Comment:
are we able to define different payload type instead of using [key: string] ?

Reply:
Object key type can only be string or number, not support other custom types like `type PayloadType = "format" | "uk" | "pk" | "idx"`. Need to figure out other better ways for both backend and frontend.